weather.py

```diff
@@ -1,109 +1,134 @@
     index = round(degrees / 45) % 8
     return directions[index]
 
 def arrow_direction(degrees):
+    if not degrees:
+        return '?'
     arrows = list('↓↙←↖↑↗→↘')
     index = round(degrees / 45) % 8
     return arrows[index]
 
-def get_weather(woeid, unit, format, timeout=None):
-    url = BASE_WEATHER_URL + urllib.parse.urlencode({
-        'q': WEATHER_QUERY.format(unit=unit, woeid=woeid),
-        'format': 'xml'
-    })
-    logging.info("Fetching %s" % url)
-    try:
-        r = requests.get(url, timeout=timeout)
-    except requests.exceptions.ConnectionError:
-        return ''
-    except requests.exceptions.Timeout:
-        return ''
-    if r.status_code != 200:
-        return 'HTTP error: %s' % r.status_code
-
-    s = BeautifulSoup(r.text, 'html.parser')
+def unix_to_hhmm(ts):
+    dt = datetime.fromtimestamp(ts)
+    return dt.strftime('%H:%M')
 
+def format_weather(obs, unit_temp, unit_speed, format_str):
     data = {}
-
-    # Unit information
-    # Prefix each unit key with 'unit_' to prevent ambiguity, e.g.
-    # 'temp' vs. 'temperature'
-    units = s.find('yweather:units')
-    data.update(('unit_' + attr, units.attrs[attr]) for attr in units.attrs)
-    # Location information
-    data.update(s.find('yweather:location').attrs)
-    # Basic conditions - simple description ("Fair", "Cloudy", etc), temperature
-    data.update(s.find('yweather:condition').attrs)
-    # Wind conditions - speed, chill, direction (degrees)
-    # This is a little bit different from the others in that the attributes
-    # are prefixed with 'wind_'.  The justification for this is that it doesn't really
-    # make sense to ask "what direction does the weather have?", as opposed to
-    # "what wind_direction does the weather have?"
-    wind = s.find('yweather:wind')
-    data.update(('wind_' + attr, wind.attrs[attr]) for attr in wind.attrs)
-    data['wind_direction_fuzzy'] = fuzzy_direction(int(data['wind_direction']))
-    data['wind_direction_arrow'] = arrow_direction(int(data['wind_direction']))
-    # Atmospheric conditions - humidity, visibility, pressure
-    data.update(s.find('yweather:atmosphere').attrs)
-    # Astronomical conditions - sunrise / sunset
-    data.update(fix_sunrise_sunset(s.find('yweather:astronomy').attrs))
-    return format.format(**data)
-
-sunrise_sunset_re = re.compile(r'(?P<hour>\d+):(?P<minute>\d+) (?P<am_pm>am|pm)')
-def fix_sunrise_sunset(attrs):
-    for key, value in attrs.items():
-        match = sunrise_sunset_re.fullmatch(value)
-        if match:
-            yield key, '{hour}:{minute:>02} {am_pm}'.format(**match.groupdict())
-        else:
-            yield key, value
+    data['unit_pressure'] = 'hPa'
+    if unit_temp == 'fahrenheit':
+        data['unit_temperature'] = 'F'
+    else:
+        data['unit_temperature'] = 'C'
+    if unit_speed == 'miles_hour':
+        data['unit_speed'] = 'mph'
+    else:
+        data['unit_speed'] = 'm/s'
```

Comment:
Wouldn't it be simpler (no need to convert) and also more user-friendly to accept "F", "C", "mph", and "m/s" on the command line instead? Nobody wants to have to spell out "fahrenheit".

Reply:
It still needs to be converted (in the other direction, to the parameter given to pyowm), but I agree that going that direction would be more intuitive for the user-facing parameters.

Reply:
Oops, I misread and thought the "m/s" value is what was being passed to pyowm.